File: swift/leet_code/1700.swift
class Solution {
    func countStudents(_ students: [Int], _ sandwiches: [Int]) -> Int {
        var students = students
        var sandwiches = sandwiches
        var i = 0
        while i < students.count {
            if students[0] == sandwiches[0] {
                students.removeFirst()
                sandwiches.removeFirst()
                i = 0
            } else {
                students.append(students.removeFirst())
                i += 1
            }
        }
        return students.count
    }
}

File: php/leet_code/1700.php
<?php

class Solution
{

    /**
     * @param Integer[] $students
     * @param Integer[] $sandwiches
     * @return Integer
     */
    public function countStudents(array $students, array $sandwiches): int
    {
        $queue = new SplQueue();
        foreach ($students as $student) {
            $queue->enqueue($student);
        }

        $stack = new SplStack();
        for ($i = count($sandwiches) - 1; $i >= 0; $i--) {
            $stack->push($sandwiches[$i]);
        }

        $count = 0;
        while (!$queue->isEmpty() && $count < count($students)) {
            $student = $queue->dequeue();
            if ($student == $stack->top()) {
                $stack->pop();
                $count = 0;
            } else {
                $queue->enqueue($student);
                $count++;
            }
        }

        return $queue->count();
    }
}
